overlord/snapstate/handlers_prereq_test.go

```diff
@@ -136,16 +136,95 @@ func (s *prereqSuite) TestDoPrereqWithBaseNone(c *C) {
 	// check that the do-prereq task added all needed prereqs
 	expectedLinkedSnaps := []string{"prereq1", "snapd"}
 	linkedSnaps := make([]string, 0, len(expectedLinkedSnaps))
+	lane := 0
 	for _, t := range chg.Tasks() {
 		if t.Kind() == "link-snap" {
 			snapsup, err := snapstate.TaskSnapSetup(t)
 			c.Assert(err, IsNil)
 			linkedSnaps = append(linkedSnaps, snapsup.InstanceName())
+		} else if t.Kind() == "prerequisites" {
+			c.Assert(t.Lanes(), DeepEquals, []int{lane})
+			lane++
 		}
 	}
+	c.Assert(lane, Equals, 3)
 	c.Check(linkedSnaps, DeepEquals, expectedLinkedSnaps)
 }
 
+func (s *prereqSuite) TestDoPrereqManyTransactional(c *C) {
+	s.state.Lock()
+
+	t := s.state.NewTask("prerequisites", "test")
+	t.Set("snap-setup", &snapstate.SnapSetup{
+		SideInfo: &snap.SideInfo{
+			RealName: "foo",
+			Revision: snap.R(33),
+		},
+		Base: "none",
+		PrereqContentAttrs: map[string][]string{
+			"prereq1": {"some-content"}, "prereq2": {"other-content"}},
+		Flags: snapstate.Flags{Transactional: true},
+	})
+	// Set lane to make sure new tasks will match this one
+	lane := 3
+	t.JoinLane(lane)
```

Comment:
Is there a reason for not using `lane := st.NewLane()` here?

Reply:
No special reason, I've changed this now to using NewLane, thanks